src/coprocessor/codec/convert.rs

```diff
@@ -243,23 +404,53 @@ fn get_valid_float_prefix<'a>(ctx: &mut EvalContext, s: &'a str) -> Result<&'a s
             valid_len = i + 1;
         }
     }
-    box_try!(ctx.handle_truncate(valid_len < s.len()));
+    ctx.handle_truncate(valid_len < s.len())?;
     if valid_len == 0 {
         Ok("0")
     } else {
         Ok(&s[..valid_len])
     }
 }
 
+fn round_int_str(num_next_dot: char, s: &str) -> Cow<'_, str> {
+    if num_next_dot < '5' {
+        return Cow::Borrowed(s);
+    }
+
+    let mut int_str = String::with_capacity(s.len());
+    match s.rfind(|c| c != '9' && c != '+' && c != '-') {
```

Comment:
Why it is testing against ` c != '9' && c != '+' && c != '-'`? In TiDB's `roundIntStr` implementation `+` and `-` is not checked at all.

Reply:
I handle the "+"/"-" in this function to make `float_str_to_int_string` more clear.